### PR TITLE
Allow the drop zone to be the drop zone instead of the entire page an…

### LINKED ISSUE #44: Override the message before "browse"
I would like to be able to override the _message property with something more specific from my app. Ex. Only ZIP files are able to be dropped it would be nice to have a message that reflects this.

### PR DESCRIPTION
**Pull request by zwilson2**
…d allow the user to override the message.

Fixes #44 and #43

### CODE REVIEW

File: d2l-file-uploader.html [hunk 1]
```diff
@@ -250,31 +264,41 @@
...
 				this.loadResources(this.resolveUrl('locales.json'));
 
-				this.listen(document, 'dragover', '__onDragOver');
-				this.listen(document, 'dragleave', '__onDragLeave');
-				this.listen(document, 'dragend', '__onDragEnd');
-				this.listen(document, 'drop', '__onDrop');
+				this._dropZone = document;
+				if (this.dropZoneOnly) {
+					this._dropZone = this.querySelector('.d2l-file-uploader-drop-zone');
+				}
```
Comment (dlockhart) [inline d2l-file-uploader.html]: Suggestion: `this._dropZone = this.dropZoneOnly ? this.querySelector(...) : document;

**Comment by dlockhart** [comment]
Could you break these into 2 separate pull requests, one that does the dropzone thing and the other for the customized message?

**Comment by dlockhart** [comment]
I'm still not convinced allowing people to override the default message is actually what we want. It seems like it's just going to lead to inconsistency in wording for users over time as the control evolves.

Maybe a better interim option might be to allow something custom to go _inside_ the control? Like this:

```
<d2l-file-uploader>Only ZIP files are permitted, or whatever</d2l-file-uploader>
```

Then that custom content just gets put right after the built-in default message. Just a thought.

**Comment by zwilson2** [comment]
This will also require an update of the current text as it says "Drag and drop a file anywhere on this screen or browse" which would be incorrect with the dropZoneOnly attribute.

**Comment by zwilson2** [comment]
This has been split into two PRs #46 and #47

**Pull request closed without merging**